vyper/codegen/function_definitions/external_function.py

```diff
@@ -123,7 +123,26 @@ def handler_for(calldata_kwargs, default_kwargs):
 
         ret.append(["goto", sig.external_function_base_entry_label])
 
-        ret = ["if", ["eq", "_calldata_method_id", method_id], ret]
+        # if there is a function whose selector is 0, it won't be distinguished
+        # from the case where nil calldata is supplied, b/c calldataload loads
+        # 0s past the end of physical calldata (cf. yellow paper).
+        # since supplying 0 calldata is expected to trigger the fallback fn,
+        # we check that calldatasize > 0, which distinguishes the 0 selector
+        # from the fallback function "selector"
+        # (equiv. to "all selectors not in the selector table").
+
+        # note: fns with trailing 0s in the selector are not checked here
+        # b/c not providing the trailing 0s in calldata is considered invalid
+        # for specifying the method id, and the contract should revert
```

Comment:
i'm not sure this comment is required anymore -- or if it is there, it should point out that calldatasize well-formedness check is added in the function body as part of ABI validation.

Reply:
"trailing 0s" is a bit confusing since it refers to the method id as opposed to the calldata. you could say, "cases where not enough calldata is supplied besides calldatasize == 0 ...".

Reply:
yeah, in the latest commit I edited this comment with your suggestions, but I am also thinking we can just do away with it (starting at note:)